src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -0,0 +1,333 @@
+                        if (membersByName.Length != 1)
+                        {
+                            // Emit diagnostic
+                            continue;
+                        }
+                        int numArguments = method.Parameters.Length;
+                        const string argumentVariableIdentifier = "testArguments";
+                        var argsAsCode = Enumerable.Range(0, numArguments).Select(i => $"{argumentVariableIdentifier}[{i}]").ToImmutableArray();
+                        testCasesBuilder.Add(new MemberDataTest(membersByName[0], new BasicTestMethod(method, alias, argsAsCode), alias, argumentVariableIdentifier));
+                        break;
+                    }
+                default:
+                    break;
+            }
+        }
+        return testCasesBuilder.ToImmutable();
+    }
+
+    private static ImmutableArray<ITestInfo> FilterForSkippedRuntime(ImmutableArray<ITestInfo> testInfos, int v, AnalyzerConfigOptionsProvider options)
+    {
+        Xunit.TestRuntimes skippedRuntimes = (Xunit.TestRuntimes)v;
+        options.GlobalOptions.TryGetValue("build_property.RuntimeFlavor", out string? runtimeFlavor);
+        if (runtimeFlavor == "Mono" && skippedRuntimes.HasFlag(Xunit.TestRuntimes.Mono))
+        {
+            return ImmutableArray<ITestInfo>.Empty;
+        }
+        else if ((string.IsNullOrEmpty(runtimeFlavor) || runtimeFlavor == "CoreCLR") && skippedRuntimes.HasFlag(Xunit.TestRuntimes.CoreCLR))
+        {
+            return ImmutableArray<ITestInfo>.Empty;
+        }
+        return testInfos;
+    }
+
+    private static ImmutableArray<ITestInfo> DecorateWithSkipOnPlatform(ImmutableArray<ITestInfo> testInfos, int v, AnalyzerConfigOptionsProvider options)
+    {
+        Xunit.TestPlatforms platformsToSkip = (Xunit.TestPlatforms)v;
+        options.GlobalOptions.TryGetValue("build_property.TargetOS", out string? targetOS);
+        Xunit.TestPlatforms targetPlatform = GetPlatformForTargetOS(targetOS);
+
+        if (platformsToSkip.HasFlag(targetPlatform))
+        {
+            // If the target platform is skipped, then we don't have any tests to emit.
+            return ImmutableArray<ITestInfo>.Empty;
+        }
+        else if (targetPlatform.HasFlag(platformsToSkip))
+        {
+            // If our target platform encompases one or more of the skipped platforms,
+            // emit a runtime platform check here.
+            Xunit.TestPlatforms platformsToEnableTest = targetPlatform & ~platformsToSkip;
+            return ImmutableArray.CreateRange(testInfos.Select(t => (ITestInfo)new PlatformSpecificTest(t, platformsToEnableTest)));
+        }
+        else
+        {
+            // The target platform is not mentioned in the attribute, just run it as-is.
+            return testInfos;
+        }
+
+        static Xunit.TestPlatforms GetPlatformForTargetOS(string? targetOS)
+        {
+            return targetOS switch
```

Comment:
I think we should `.ToLower()` this since we might not always get the correct casing from msbuild.

Reply:
```suggestion
            return targetOS.ToLower() switch
```

Reply:
Nit: `ToLowerInvariant`